pkg/deploy/deploy.go

```diff
@@ -113,7 +159,11 @@ func (d *deployer) PreDeploy(ctx context.Context) (rpServicePrincipalID string,
 			return "", err
 		}
 
-		deployment, err = d.deployments.Get(ctx, d.resourceGroup, deploymentName)
+		deployment, err := d.deployments.Get(ctx, d.resourceGroup, deploymentName)
+		if err != nil {
+			return "", err
+		}
+		return deployment.Properties.Outputs.(map[string]interface{})["rpServicePrincipalId"].(map[string]interface{})["value"].(string), nil
```

Comment:
this is not an improvement.  Perhaps putting all of this scope into a separate function, say "_deployManageIdentity" will help.

Reply:
didnt understood. It's already in similar function...